docs/guides/udc.md

```diff
@@ -72,3 +72,28 @@ PRINT_VAR:
     COMMAND
     RUN echo "$a_global_var"
 ```
+
+## Targets vs UDCs
+
+Earthly targets and UDCs are Earthly's core primitives for organizing build recipes. They encapsulate build logic, and from afar they look pretty similar. However, the use-cases for each are vastly different.
+
+In general, targets are used to produce specific build results, while UDCs are used as a way to reuse build logic, when certain commands are repeated in multiple places. As a real-world analogy, targets are more like factories, while UDCs are more like components that are used to put together factories.
```

Comment:
What about adding "UDCs work like function or methods call in an imperative programming language (c, javascript, python). Much like function calls it's helpful to imagine UDCs being executed by being inlined into the call site but in a separate variable scope"

Reply:
Good idea